File: 0240-search-a-2d-matrix-ii/0240-search-a-2d-matrix-ii.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        m = len(matrix)
        n = len(matrix[0])
        row = 0
        column = n - 1

        while column >= 0 and row < m:
            if matrix[row][column] == target:
                return True
            elif matrix[row][column] < target:
                row = row + 1
            else:
                column = column - 1
        return False

File: 0240-search-a-2d-matrix-ii/0240-search-a-2d-matrix-ii.java
class Solution {
    public boolean searchMatrix(int[][] matrix, int target) {
        int m = matrix.length;
        int n = matrix[0].length;

        int targetRow = m - 1, targetCol = 0;

        while (targetRow >= 0 && targetCol < n) {
            if (matrix[targetRow][targetCol] == target) {
                return true;
            }

            if (matrix[targetRow][targetCol] > target) {
                targetRow--;
            } else {
                targetCol++;
            }
        }

        return false;
    }
}